apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java

```diff
@@ -1,8 +1,70 @@
 package com.apollographql.apollo;
 
+/**
+ * CustomTypeAdapter class represents the adapter for converting custom scalar types to Java objects.
+ *
+ * <h3>Defining a custom scalar type adapter</h3>
+ *
+ * The GraphQL specification includes the following default scalar types: <b>String, Int, Float and Boolean</b>. But
+ * often use cases arise when support for custom scalar types is required and for those cases this class should be used
+ * to customize conversion. Here's an example of a type adapter for scalar type {@link java.util.Date}:
+ * <pre>
+ *   {@code
+ *
+ *  CustomTypeAdapter<Date> dateCustomTypeAdapter = new CustomTypeAdapter<Date>() {
+ *                @Override
+ *                public Date decode(String value) {
+ *                  try {
+ *                    return ISO8601_DATE_FORMAT.parse(value);
+ *                  } catch (ParseException e) {
+ *                    throw new RuntimeException(e);
+ *                  }
+ *                }
+ *                @Override
+ *                public String encode(Date value) {
+ *                  return ISO8601_DATE_FORMAT.format(value);
+ *                }
+ *          };
+ * }
+ * </pre>
+ * To use a custom type adapter with Apollo, you must first define the mapping in the build.gradle file:
+ * <pre>   {@code
+ *  apollo {
+ *        generateClasses = true
+ *        customTypeMapping {
+ *        DateTime = "java.util.Date"
```

Comment:
We should mention that sometimes when name of the type is reserved keyword in groovy you can use Map notation, for example:
```
customTypeMapping['URL'] = "String"
```
where URL is reserved keyword in groovy and you can't use it (will get compilation error):
```
URL = "String"
```